modules/table/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceDistributedTest.java

```diff
@@ -369,6 +370,8 @@ void before(TestInfo testInfo) throws Exception {
     @AfterEach
     void after() {
         nodes.forEach(Node::stop);
+
+        MockUtil.clearAllCaches();
```

Comment:
1. What's the rule of thumb to define wether clearAllCaches is required or not?
2. Is it required to call it in AfterEach or AfterAll will also do the trick?
3. What's the disadvantage of calling clearAllCaches()? In other words, why not to call it in AfterAll within every base test. E.g. we may call it BaseIgniteAbstractTest.
4. What's the difference between clearInlineMocks and clearAllCaches? Why do we need to call both?

Reply:
https://issues.apache.org/jira/browse/IGNITE-23956
Described the issue here and added TODOs.

Reply:
The description of https://issues.apache.org/jira/browse/IGNITE-23956 seems a bit messy. Do you mean that besides clearInlineMocks() it's also required to call clearAllCaches within @AfterAll?

Reply:
Anyway, all questions above still lacks answers.

Reply:
clearAllCaches() includes clearInlineMocks(). The last one is necessary to avoid leak,

Reply:
1. What's the rule of thumb to define wether clearAllCaches is required or not?

Reply:
2. Is it required to call it in AfterEach or AfterAll will also do the trick?

Reply:
3. What's the disadvantage of calling clearAllCaches()? In other words, why not to call it in AfterAll within every base test. E.g. we may call it BaseIgniteAbstractTest.

Reply:
4. What's the difference between clearInlineMocks and clearAllCaches? Why do we need to call both?
Got it. You've answered it.

Reply:
By questions:
1. Invoke clearInlineMocks() after any test (in the case were the mockito do not use the method do nothing), is case where we are not going to use created mocks in the next test.
2. Depends on scenario, look at poit 1.
3. Many test use MockitoExtension. The extention manages mocks independently.

Reply:
Still not clear.